pypeit/core/wavecal/wv_fitting.py

```diff
@@ -178,20 +279,30 @@ def iterative_fitting(spec, tcent, ifit, IDs, llist, disp,
             flg_continue = False
         # Fit with rejection
         xfit, yfit, wfit = tcent[ifit], all_ids[ifit], weights[ifit]
-        mask, fit = utils.robust_polyfit(xfit/xnspecmin1, yfit, n_order, function=func, sigma=sigrej_first,
-                                         minx=fmin, maxx=fmax, verbose=verbose, weights=wfit)
+        maxiter = xfit.size - n_order - 2
+        #
+        if xfit.size == 0:
+            msgs.warn("All points rejected !!")
+            return None
+        # Fit
+        pypeitFit = fitting.robust_fit(xfit/xnspecmin1, yfit, n_order, function=func, maxiter=maxiter,
+                                       lower=sigrej_first, upper=sigrej_first, maxrej=1, sticky=True,
+                                       minx=fmin, maxx=fmax, weights=wfit)
+        # Junk fit?
+        if pypeitFit is None:
+            msgs.warn("Bad fit!!")
+            return None
 
-        rms_ang = utils.calc_fit_rms(xfit[mask == 0]/xnspecmin1, yfit[mask == 0], fit, func, minx=fmin, maxx=fmax,
-                                     weights=wfit[mask == 0])
+        rms_ang = pypeitFit.calc_fit_rms(apply_mask=True)
         rms_pix = rms_ang/disp
         if verbose:
             msgs.info('n_order = {:d}'.format(n_order) + ': RMS = {:g}'.format(rms_pix))
 
         # Reject but keep originals (until final fit)
-        ifit = list(ifit[mask == 0]) + sv_ifit
+        ifit = list(ifit[pypeitFit.gpm == 1]) + sv_ifit
```

Comment:
hmmm... is the gpm supposed to boolean, anyway? It might be good to make it boolean after being read in from file?

Reply:
For now, gpm is an `int` for I/O.  Hope to make it `bool` someday soon..